core/domain/src/commonMain/kotlin/com/mifos/core/domain/useCases/GetCheckerInboxBadgesUseCase.kt

```diff
@@ -21,10 +21,32 @@ import kotlinx.coroutines.flow.zip
 class GetCheckerInboxBadgesUseCase(
     private val repository: CheckerInboxTasksRepository,
 ) {
-    operator fun invoke(): Flow<DataState<Pair<Int, Int>>> =
+    operator fun invoke(): Flow<DataState<Pair<Int, Int>>> = combine(
+        repository.getCheckerTaskList(),
+        repository.getRescheduleLoansTaskList(),
+    ) { checkerTaskState, rescheduleTaskState ->
+        when {
+            checkerTaskState is DataState.Loading || rescheduleTaskState is DataState.Loading -> {
+                DataState.Loading
+            }
+
+            checkerTaskState is DataState.Success && rescheduleTaskState is DataState.Success -> {
+                val checkerTaskSize = checkerTaskState.data.size
+                val rescheduleTaskSize = rescheduleTaskState.data.size
+                DataState.Success(checkerTaskSize to rescheduleTaskSize)
+            }
 
-        repository.getCheckerTaskList()
-            .zip(repository.getRescheduleLoansTaskList()) { checkerTasks, rescheduleTasks ->
-                DataState.Success(Pair(checkerTasks.data!!.size, rescheduleTasks.data!!.size))
+            else -> {
+                val errors = listOfNotNull(
+                    (checkerTaskState as? DataState.Error)?.exception,
+                    (rescheduleTaskState as? DataState.Error)?.exception,
+                )
+                DataState.Error(combineErrors(errors))
             }
+        }
+    }
+
+    fun combineErrors(errors: List<Throwable>): Throwable {
```

Comment:
it is fine but just wanted to say because this `combineErrors` is only used in this usecase and is single line I think it would be better to move it directly inside the `DataStateError()`
```
DataState.Error(
    exception = Throwable(errors.joinToString("\n") { it.message ?: "Unknown error" })
)
```